overcast/runner/__init__.py

```diff
@@ -413,16 +474,36 @@ def provision_step(self, details):
         for base_node_name, node_info in stack['nodes'].items():
             def _create_node(base_name):
                 node_name = self.add_prefix(base_name)
-                self.nodes[base_name] = self.create_node(node_name, node_info,
-                                                         keypair=keypair_name,
-                                                         userdata=userdata)
+                self.nodes[base_name] = Node(node_name, node_info,
+                                                        keypair=keypair_name,
+                                                        userdata=userdata,
+                                                        runner=self)
+                self.nodes[base_name].build()
 
             if 'number' in node_info:
                 count = node_info.pop('number')
                 for idx in range(1, count+1):
-                    _create_node('%s%d' % (base_node_name, idx))
+                    node_name = '%s%d' % (base_node_name, idx)
+                    _create_node(node_name)
+                    pending_nodes.add(node_name)
             else:
                 _create_node(base_node_name)
+                pending_nodes.add(base_node_name)
+
+        while pending_nodes:
+            time.sleep(5)
```

Comment:
I think I'd put "done = set()" right here instead of above (line 447). The only thing we use done() for is to remove nodes from pending_nodes after they become ACTIVE, so we might as well reset done itself for every loop.